bundles/mapping/heatmap/HeatmapDialog.js

```diff
@@ -174,23 +181,25 @@ Oskari.clazz.define('Oskari.mapframework.bundle.heatmap.HeatmapDialog',
             this.colors = [];
             this.colors.push("#FFFFFF");
             var me = this,
-                element = jQuery(content).find("input:checked"),
-                colorBoxes = jQuery(element[0].parentElement).find('.color-box');
-            _.forEach(colorBoxes, function (colorBox) {
-                me.colors.push(jQuery(colorBox).attr("color"));
-            });
+                element = jQuery(content).find("input:checked");
+
+            if (element.attr('id') === "customTheme") {
+                this.colors = this.colors.concat(this.getColorSetup());
+            }
+            else {
+                var colorBoxes = jQuery(element[0].parentElement).find('.color-box');
+                colorBoxes.forEach(function (colorBox) {
```

Comment:
`colorBoxes` is a jQuery selection, not an Array. It does not have `forEach()` method.